src/figuresClasses.ts

```diff
@@ -1,11 +1,84 @@
-export interface Figure {}
+type Shapes = 'circle' | 'triangle' | 'rectangle';
+type Colors = 'red' | 'green ' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  readonly shape = 'triangle';
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Incorrect number');
```

Comment:
The error message 'Incorrect number' is not very descriptive. It would be better to specify which parameter is incorrect and why, for example 'Triangle side length must be greater than 0'.